Source/Charts/Charts/PieRadarChartViewBase.swift

```diff
@@ -672,23 +672,8 @@ open class PieRadarChartViewBase: ChartViewBase
         
         _velocitySamples.append(AngularVelocitySample(time: currentTime, angle: angleForPoint(x: touchLocation.x, y: touchLocation.y)))
         
-        // Remove samples older than our sample time - 1 seconds
-        var i = 0, count = _velocitySamples.count
-        while (i < count - 2)
-        {
-            if currentTime - _velocitySamples[i].time > 1.0
-            {
-                _velocitySamples.remove(at: 0)
-                i -= 1
-                count -= 1
-            }
-            else
-            {
-                break
-            }
-            
-            i += 1
-        }
+        // Remove samples older than our sample time - 1 seconds        
+        _velocitySamples = Array(_velocitySamples.dropLast(2).drop { currentTime - $0.time > 1.0 })
```

Comment:
```
_velocitySamples = Array(
    _velocitySamples
        .dropLast(2)
        .drop { currentTime - $0.time > 1.0 }
)
```

Reply:
Done.

Reply:
@jjatie I found a bug in my pull request and I want to fix it.

Reply:
Go ahead. I don’t see the bug though.

Reply:
Oh, we don’t need to `dropLast(2)` only `drop(while:)`

Reply:
I'd prefer to use `removeAll(where:)` in this case. They are both O(n) operations, but we don't need to allocate a new `Array`. i.e.:
```
_velocitySamples.removeAll { currentTime - $0.time > 1.0 }
```
To be more true to the original implementation (although I don't think it's necessary, we can do:
```
_velocitySamples
    .dropLast()
    .removeAll { currentTime - $0.time > 1.0 }
```